### PR TITLE
Refactor/1019/containers checkout-form, self-pickup, header-profile

### PR DESCRIPTION
**Pull request by AnastasiaKhukhra**
## Description

#1019  Applying i18next for checkout-form, self-pickup, footer-links, footer-lists, header-profile

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/containers/footer-links/tests/footer-links.spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import Enzyme, { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import FooterLinks from '../footer-links';
+
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (maarekua) [inline src/containers/footer-links/tests/footer-links.spec.js:6]: You can remove adapter enzyme and shallow.

Comment (maarekua) [reply src/containers/footer-links/tests/footer-links.spec.js:6]: Cause after reconfiguration those imports are redunant, same for all other tests

File: src/configs/index.js [hunk 2]
```diff
@@ -546,3 +546,22 @@ export const ITEMS_PER_PAGE = [
     value: 30
   }
 ];
+
+export const CHECKOUT_INPUT_FIELD = {
```
Comment (maarekua) [inline src/configs/index.js:550]: I think we can remove this consts, cause attribute equals to value all the timte.

File: src/containers/checkout/checkout-form/checkout-form.js [hunk 3]
```diff
@@ -50,57 +42,55 @@ import {
...
 
   const { dirty, values, handleSubmit, handleChange, setFieldValue, touched, errors, resetForm } =
     useFormik({
       validationSchema: validationSchema(deliveryType, t),
       initialValues,
 
       onSubmit: (data) => {
-        data.paymentMethod === CHECKOUT_PAYMENT[language].card
-          ? dispatch(addPaymentMethod(CHECKOUT_PAYMENT[language].card)) &&
+        data.paymentMethod === t('checkout.checkoutPayment.card')
```
Comment (maarekua) [inline src/containers/checkout/checkout-form/checkout-form.js:83]: Translations are only for translating values, not for the ternaries or maps etc. I think we need to refactor this and not use translations in logic elements.

File: src/containers/checkout/checkout-form/checkout-form.js [hunk 4]
```diff
@@ -192,24 +182,24 @@ const CheckoutForm = ({ language, isLightTheme, currency, cartItems, deliveryTyp
...
-                  label={CHECKOUT_TEXT_FIELDS[language].paymentMethod}
+                  label={t('checkout.checkoutTextFields.paymentMethod')}
                   className={styles.paymentSelect}
                   data-cy={CHECKOUT_INPUT_FIELD.paymentMethod}
                   name={CHECKOUT_INPUT_FIELD.paymentMethod}
                   value={values.paymentMethod}
                   onChange={handleChange}
                 >
-                  {Object.values(CHECKOUT_PAYMENT[language]).map((value) => (
+                  {Object.values(getCheckoutPayment).map((value) => (
```
Comment (maarekua) [inline src/containers/checkout/checkout-form/checkout-form.js:202]: Using map on translation values is not a good idea.

File: src/containers/checkout/checkout-form/delivery/self-pickup/self-pickup.js [hunk 5]
```diff
@@ -1,34 +1,35 @@
...
+import { useTranslation } from 'react-i18next';
 import { useStyles } from './self-pickup.styles';
-import { CHECKOUT_TITLES, SCHEDULE } from '../../../../../translations/checkout.translations';
 
-const SelfPickup = ({ isLightTheme, language }) => {
+const SelfPickup = ({ isLightTheme }) => {
   const styles = useStyles({ isLightTheme });
+  const { t } = useTranslation();
+  const getSchedule = t('checkout.schedule', { returnObjects: true });
+  const schedule = Object.values(getSchedule).map((value) =>
```
Comment (maarekua) [inline src/containers/checkout/checkout-form/delivery/self-pickup/self-pickup.js:10]: Same here.

**Review by Skrekliam — APPROVED** [review]

File: src/containers/checkout/checkout-form/checkout-form.js [hunk 6]
```diff
@@ -47,50 +39,49 @@ import {
...
 
   const { dirty, values, handleSubmit, handleChange, setFieldValue, touched, errors, resetForm } =
     useFormik({
       validationSchema: validationSchema(deliveryType, t),
       initialValues,
 
       onSubmit: (data) => {
-        data.paymentMethod === CHECKOUT_PAYMENT[language].card
-          ? dispatch(addPaymentMethod(CHECKOUT_PAYMENT[language].card)) &&
+        data.paymentMethod === checkoutPayMethod[0].label
```
Comment (maarekua) [inline src/containers/checkout/checkout-form/checkout-form.js:83]: Array to object.

**Review by Skrekliam — APPROVED** [review]

**Pull request merged**